lstchain/image/cleaning/cleaning.py

```diff
@@ -0,0 +1,140 @@
+import tables
+import numpy as np
+
+from lstchain.io.io import dl1_params_tel_mon_ped_key, dl1_params_tel_mon_cal_key
+
+
+def tailcuts_clean_with_pedestal_threshold(
+    geom,
+    image,
+    interleaved_pedestal_thresh,
+    picture_thresh=7,
+    boundary_thresh=5,
+    keep_isolated_pixels=False,
+    min_number_picture_neighbors=0,
+):
+
+    """ Modified tailcuts_clean method from ctapipe.
+    Clean an image by selection pixels that pass a two-threshold
+    tail-cuts procedure.  The picture and boundary thresholds are
+    defined with respect to the pedestal dispersion. The picture threshold have
+    additional condition: picture threshold > threshold from interleaved
+    pedestal events. All pixels that have a signal higher than the picture
+    threshold will be retained, along with all those above the boundary
+    threshold that are neighbors of a picture pixel.
+
+    To include extra neighbor rows of pixels beyond what are accepted, use the
+    `ctapipe.image.dilate` function.
+    Parameters
+    ----------
+    geom: `ctapipe.instrument.CameraGeometry`
+        Camera geometry information
+    image: array
+        pixel values
+    interleaved_pedestal_thresh: array
+        additional picture threshold calculated using interleaved events
+    picture_thresh: float or array
+        threshold above which all pixels are retained
+    boundary_thresh: float or array
+        threshold above which pixels are retained if they have a neighbor
+        already above the picture_thresh
+    keep_isolated_pixels: bool
+        If True, pixels above the picture threshold will be included always,
+        if not they are only included if a neighbor is in the picture or
+        boundary
+    min_number_picture_neighbors: int
+        A picture pixel survives cleaning only if it has at least this number
+        of picture neighbors. This has no effect in case keep_isolated_pixels is True
+    Returns
+    -------
+    A boolean mask of *clean* pixels.  To get a zero-suppressed image and pixel
+    list, use `image[mask], geom.pix_id[mask]`, or to keep the same
+    image size and just set unclean pixels to 0 or similar, use
+    `image[~mask] = 0`
+    """
+
+    pixels_above_picture = np.logical_and(image>= picture_thresh,
```

Comment:
You do not need your own function to do this.

You can just use `picture_thresh = np.clip(pedestal_threshhold, picture_threshold, None)` and use that as `picture_threshold` for the normal image cleaning

Reply:
Good idea, thank you